cohort-engine/src/main/java/com/ibm/cohort/engine/measure/CDMMeasureEvaluation.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * (C) Copyright IBM Corp. 2020, 2020
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package com.ibm.cohort.engine.measure;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+
+import org.hl7.fhir.r4.model.Measure;
+import org.hl7.fhir.r4.model.MeasureReport;
+import org.hl7.fhir.r4.model.codesystems.MeasureScoring;
+import org.opencds.cqf.common.evaluation.MeasurePopulationType;
+import org.opencds.cqf.cql.engine.data.DataProvider;
+import org.opencds.cqf.cql.engine.execution.Context;
+import org.opencds.cqf.cql.engine.runtime.Interval;
+import org.opencds.cqf.r4.evaluation.MeasureEvaluation;
+
+/**
+ * Implementation of measure evaluation logic for the IBM Common Data Model IG
+ * Patient Quality Measure profile.
+ */
+public class CDMMeasureEvaluation {
+
+	public static final String CARE_GAP = "care-gap";
+	public static final String CDM_CODE_SYSTEM_MEASURE_POPULATION_TYPE = "http://ibm.com/fhir/cdm/CodeSystem/measure-population-type";
+
+	/**
+	 * Helper for collecting and indexing the various standard population types from
+	 * base FHIR and their count values so that they can easily be referenced in the
+	 * business logic.
+	 */
+	public static final class StandardReportResults extends HashMap<MeasurePopulationType, Boolean> {
+
+		private static final long serialVersionUID = 1L;
+
+		public boolean inInitialPopulation() {
+			return inPopulation(MeasurePopulationType.INITIALPOPULATION);
+		}
+
+		public boolean inDenominator() {
+			return inPopulation(MeasurePopulationType.DENOMINATOR);
+		}
+
+		public boolean inDenominatorExclusion() {
+			return inPopulation(MeasurePopulationType.DENOMINATOREXCLUSION);
+		}
+
+		public boolean inDenominatorException() {
+			return inPopulation(MeasurePopulationType.DENOMINATOREXCEPTION);
+		}
+
+		public boolean inNumerator() {
```

Comment:
Not sure if these method names reflect anything in cqf-ruler - if not, slight preference for renaming to something like "meetsNumeratorCriteria" or anything to reflect the fact that we're not trying to do the measure logic here (the patient is not necessarily in the numerator without evaluating the other populations).  

If I'm the only one who reads it that way then feel free to ignore, but I did spend an extra few minutes looking into this class trying to figure out if those methods were used to determine population results (which it seems not).

Reply:
These are entirely my construction. I let the cqf-ruler do its thing for the standard populations and then I need to know after that runs whether or not a care-gap should be evaluated. The helper class checks the count for each population and then keeps track using a boolean whether they are in that population or not. I'm not sure "meets criteria" matches those semantics. For instance, if a patient meets both the numerator criteria and the numerator exception criteria, the are removed from the numerator (inNumerator = false and inNumeratorException = true). I'm not firm on this, so if you still think it makes sense to change the names I can do so.

Reply:
Hmm ok - I didn't read the inNumerator() as returning false if inNumeratorException() was true, but I understand better now.